src/encoded/static/components/cart/cart.js

```diff
@@ -93,53 +97,182 @@ CartSearchResults.defaultProps = {
+    return (
+        <div className="cart-facet-term__magnitude">
+            <i className="icon icon-circle" style={{ color: magnitudeColor }} />
+        </div>
+    );
+};
+
+FacetTermMagnitude.propTypes = {
+    /** Number of items this facet term indicates */
+    termCount: PropTypes.number.isRequired,
+    /** Maximum count value among all terms in this facet */
+    maxTermCount: PropTypes.number.isRequired,
+};
+
+
+/**
+ * Display one term of a facet.
  */
 class FacetTerm extends React.Component {
     constructor() {
         super();
-        this.handleTermSelect = this.handleTermSelect.bind(this);
+        this.handleTermClick = this.handleTermClick.bind(this);
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+    }
+
+    /**
+     * Called when user clicks a term within a facet.
+     */
+    handleTermClick() {
+        this.props.termClickHandler(this.props.term);
     }
 
     /**
-     * Called when a file format is selected from the facet.
+     * Called when user types a key while focused on a facet term. If the user types a space or
+     * return we call the term click handler -- needed for a11y because we have a <div> acting as a
+     * button instead of an actual <button>.
+     * @param {object} e React synthetic event
      */
-    handleTermSelect() {
-        this.props.termSelectHandler(this.props.term);
+    handleKeyDown(e) {
+        if (e.keyCode === 13 || e.keyCode === 32) {
+            e.preventDefault();
+            this.props.termClickHandler(this.props.term);
+        }
     }
 
     render() {
-        const { term, termCount, totalTermCount, selected } = this.props;
-        const barStyle = {
-            width: `${Math.ceil((termCount / totalTermCount) * 100)}%`,
-        };
+        const { term, termCount, maxTermCount, selected } = this.props;
         return (
-            <li className={`facet-term${selected ? ' selected' : ''}`}>
-                <button id={`facet-term-${term}`} onClick={this.handleTermSelect} aria-label={`${term} with count ${termCount}`}>
-                    <div className="facet-term__item">
-                        <div className="facet-term__text"><span>{term}</span></div>
-                        <div className="facet-term__count">
-                            {termCount}
-                        </div>
-                        <div className="facet-term__bar" style={barStyle} />
-                    </div>
-                </button>
+            <li className="cart-facet-term">
+                <div
+                    className="cart-facet-term__item"
+                    role="button"
+                    tabIndex="0"
+                    id={`cart-facet-term-${term}`}
+                    onKeyDown={this.handleKeyDown}
+                    onClick={this.handleTermClick}
+                    aria-pressed={selected}
+                    aria-label={`${termCount} files ${term}`}
```

Comment:
I had it more detailed until a very late commit because, since aria-labels are read as speech, I thought they should be shorter so a visually impaired person doesn’t get annoyed. These are for individual facet terms that could get to be a very long list. I haven’t been able to find a general philosophy about aria-labels though, probably because it depends on the situation.

Right now with Voice Over for macOS, it says, “Twelve hundred twenty-two files alignments toggle button” or “Twelve hundred twenty-two files alignments selected toggle button” if the user had selected it. Other text-to-speech engines could say it differently.

Reply:
I can definitely see the benefit to keeping it short, I just thought '1222 files alignments' was confusing. Maybe just adding a preposition 'files of alignments' would make it a little more clear??

Reply:
I reversed the words, so now it says things like, “Twenty-three alignments files selected toggle button” (if it’s selected).